train_and_save.py

```diff
@@ -0,0 +1,165 @@
+
+    # Define two stacked LSTM cells (two recurrent layers deep) with tensorflow
+    lstm_cell_1 = tf.contrib.rnn.BasicLSTMCell(step.hyperparams['n_hidden'], forget_bias=1.0, state_is_tuple=True)
+    lstm_cell_2 = tf.contrib.rnn.BasicLSTMCell(step.hyperparams['n_hidden'], forget_bias=1.0, state_is_tuple=True)
+    lstm_cells = tf.contrib.rnn.MultiRNNCell([lstm_cell_1, lstm_cell_2], state_is_tuple=True)
+
+    # Get LSTM cell output
+    outputs, states = tf.contrib.rnn.static_rnn(lstm_cells, _X, dtype=tf.float32)
+
+    # Get last time step's output feature for a "many-to-one" style classifier,
+    # as in the image describing RNNs at the top of this page
+    lstm_last_output = outputs[-1]
+
+    # Linear activation
+    return tf.matmul(lstm_last_output, weights['out']) + biases['out']
+
+
+def create_optimizer(step: TensorflowV1ModelStep):
+    return tf.train.AdamOptimizer(learning_rate=step.hyperparams['learning_rate'])
+
+
+def create_loss(step: TensorflowV1ModelStep):
+    # Loss, optimizer and evaluation
+    # L2 loss prevents this overkill neural network to overfit the data
+    l2 = step.hyperparams['lambda_loss_amount'] * sum(tf.nn.l2_loss(tf_var) for tf_var in tf.trainable_variables())
+
+    # Softmax loss
+    return tf.reduce_mean(
+        tf.nn.softmax_cross_entropy_with_logits(
+            labels=step['expected_outputs'],
+            logits=step['output']
+        )
+    ) + l2
+
+
+class HumanActivityRecognitionPipeline(Pipeline):
+    N_HIDDEN = 32
+    N_STEPS = 128
+    N_INPUTS = 9
+    LAMBDA_LOSS_AMOUNT = 0.0015
+    LEARNING_RATE = 0.0025
+    N_CLASSES = 6
+    BATCH_SIZE = 1500
+    EPOCHS = 14
+
+    def __init__(self):
+        super().__init__([
+            OutputTransformerWrapper(OneHotEncoder(nb_columns=self.N_CLASSES, name='one_hot_encoded_label')),
+            FormatData(n_classes=self.N_CLASSES),
+            TensorflowV1ModelStep(
+                create_graph=create_graph,
+                create_loss=create_loss,
+                create_optimizer=create_optimizer
+            ).set_hyperparams(
+                HyperparameterSamples({
+                    'n_steps': self.N_STEPS,  # 128 timesteps per series
+                    'n_inputs': self.N_INPUTS,  # 9 input parameters per timestep
+                    'n_hidden': self.N_HIDDEN,  # Hidden layer num of features
+                    'n_classes': self.N_CLASSES,  # Total classes (should go up, or should go down)
+                    'learning_rate': self.LEARNING_RATE,
+                    'lambda_loss_amount': self.LAMBDA_LOSS_AMOUNT,
+                    'batch_size': self.BATCH_SIZE
+                })
+            )
+        ])
+
+
+def accuracy_score_classification(data_inputs, expected_outputs):
+    accuracy = np.mean(np.argmax(data_inputs, axis=1) == np.argmax(expected_outputs, axis=1))
+    return accuracy
+
+
+def main():
+    pipeline = DeepLearningPipeline(
+        HumanActivityRecognitionPipeline(),
+        validation_size=0.15,
```

Comment:
Was the original project using validation data, or only train/test? I'm tempted to remove validation data here to leave the original example untouched. The simplicity was part of its success.